src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -13,15 +13,33 @@ namespace UrbanAirship.NETStandard
 {
     public delegate void DeepLinkHandler(string deepLink);
 
-    public class Airship : Java.Lang.Object, IDeepLinkListener, IAirship
+    public class Airship : Java.Lang.Object, IDeepLinkListener, IAirship, IInboxListener
     {
-        private static Airship sharedAirship = new Airship();
+        private static Lazy<Airship> sharedAirship = new Lazy<Airship>(() =>
+        {
+            Airship instance = new Airship();
+            instance.Init();
+            return instance;
+        });
+
+        private void Init()
+        {
+            //Adding Inbox updated listener
+            EventHandler handler = OnMessageCenterUpdated;
+            if (handler != null)
+            {
+                handler(this, EventArgs.Empty);
+            }
+            MessageCenterClass.Shared().Inbox.AddListener(this);
```

Comment:
You should only add the listener here, the handler will always be empty at this point